### ISSUE TITLE
`docker compose build <service>` builds `service`'s `depends_on` services too

### ISSUE THREAD

**Issue by tim-timman** [docker/compose#8587]
## Description of the issue

Targeting a build of a specific service builds other's it `depends_on`.

## Context information (for bug reports)

- [x] Using Compose V2 `docker compose ...`
- [ ] Using Compose V1 `docker-compose ...`

**Output of `docker(-)compose version`**
```sh
# docker-compose version
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020

# docker compose version
Docker Compose version v2.0.0-rc.1
```

**Output of `docker version`**
```
Client:
 Cloud integration: 1.0.17
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:55:20 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:10 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```yml
services:
  dependent-service:
    build:
      context: /private/tmp/docker-build-bug
      dockerfile: Dockerfile
    depends_on:
      main-service:
        condition: service_started
  main-service:
    build:
      context: /private/tmp/docker-build-bug
      dockerfile: Dockerfile
version: '3.9'
```

## Steps to reproduce the issue

1. Create these files:  
    ```dockerfile
    # Dockerfile

    FROM scratch
    ```
    ```yml
    # compose.yml

    services:
      main-service:
        build:
          context: .
          dockerfile: Dockerfile
    
      dependent-service:
        build:
          context: .
          dockerfile: Dockerfile
        depends_on:
          - main-service
    ```
2. Run `docker compose build depenent-service`

### Observed result
It builds _both_ `main-service` and `dependent-service`, even though the dependency should be only at runtime.

### Expected result
Only `dependent-service` should be built. Compare it with targeting only `main-service`.

### Stacktrace / full error message

```
#1 [docker-build-bug_main-service internal] load build definition from Dockerfile
#1 sha256:5c9def5cc28366399a64fcb2e421670725e42456b93ae2825487ece48a382032
#1 transferring dockerfile: 31B done
#1 DONE 0.0s

#2 [docker-build-bug_dependent-service internal] load build definition from Dockerfile
#2 sha256:116af94184a4e42899557eb94f11a981cfeb3092717370a24dd096052be60bb2
#2 transferring dockerfile: 31B 0.0s done
#2 DONE 0.1s

#3 [docker-build-bug_main-service internal] load .dockerignore
#3 sha256:a920cfc89d7e2768d6afbbeb9a7fe0aa1fe963d80670eb35d8d4c5a0db724db9
#3 transferring context: 2B 0.0s done
#3 DONE 0.0s

#4 [docker-build-bug_dependent-service internal] load .dockerignore
#4 sha256:bd90a0b4d5b129deac39a10ae4b401d35bcaa323c31551adc836d5fdc423c197
#4 transferring context: 2B 0.0s done
#4 DONE 0.0s

#5 [docker-build-bug_main-service] exporting to image
#5 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#5 writing image sha256:71de1148337f4d1845be01eb4caf15d78e4eb15a1ab96030809826698a5b7e30 0.1s done
#5 naming to docker.io/library/docker-build-bug_main-service done
#5 naming to docker.io/library/docker-build-bug_dependent-service done
#5 DONE 0.1s
```

## Additional information

`docker-compose build dependent-service` does the right thing: building only `dependent-service`.

OS version / distribution, `docker-compose` install method, etc.
 - macOS Catalina Version 10.15.7
 - Docker Desktop Version 3.6.0 (67351)

**Comment by ndeloof** [comment]
This issue has been fixed already in v2.0.0-rc.3
(https://github.com/docker/compose-cli/pull/2044)

**Comment by tim-timman** [comment]
@ndeloof Thanks. My searches didn't find that issue: all these different repositories is a mess. And their "responsibilities" don't stay the same... Having been burned by updates breaking things before, I haven't kept it up-to-date; guess I'll have to take my changes this time.
